### PR TITLE
Improve the player client side interpolation

### PR DESCRIPTION
**Pull request by hubastard**
I've improved the `RemotePlayerMovement` client side interpolation algorithm to make sure that we still have smooth player movement even with high latency connection.

With this implementation, during my tests, I had still pretty decent movement smoothness even in the 200-300ms latency range.

I've also added some default connection latency and drop packet simulation for a "standard" remote connection to make sure that we have a realistic testing environment.

### CODE REVIEW

File: NebulaModel/Utils/TimeUtils.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+
+namespace NebulaModel.Utils
+{
+    public class TimeUtils
+    {
+        static readonly DateTime UNIX_EPOCH = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+
+        public static long CurrentUnixTimestampMilliseconds()
```
Comment (Mushroom) [inline NebulaModel/Utils/TimeUtils.cs:9]: `DateTimeOffset.Now.ToUnixTimeSeconds()` now exists :)

Bit of an easier option than doing all this.

File: NebulaServer/Server.cs [hunk 2]
```diff
@@ -24,6 +24,11 @@ public Server()
             server = new NetManager(this)
             {
                 AutoRecycle = true,
+                // Settings below only works in DEBUG builds
+                SimulateLatency = true,
```
Comment (Mushroom) [inline NebulaServer/Server.cs:28]: Might be better to gate this regardless of whether it only applies to debug or not, as it could confuse us elsewhere.

**Review by Mushroom — CHANGES_REQUESTED** [review]

File: NebulaServer/Server.cs (hunk 2 above)
Comment (hubastard) [reply NebulaServer/Server.cs:28]: Yeah sure

File: NebulaModel/Utils/TimeUtils.cs (hunk 1 above)
Comment (hubastard) [reply NebulaModel/Utils/TimeUtils.cs:9]: Doesn't exist since we are in 3.5 :(

**Pull request merged**